### PR TITLE
[snmp_listener] Update snmp listener config template

### PR DESCRIPTION
**Pull request by AlexandreYang**
### What does this PR do?

Update snmp listener config template

### Motivation

Some configs where missing from config.go and config template.

### Describe how to test your changes

- Check no warning is raised when configs from configs.go changes are used
- Check new configs are present in datadog.yaml

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -655,6 +668,12 @@ api_key:
     #
     # loader: core
 
+    ## @param min_collection_interval - number - optional - default: 15
+    ## This changes the collection interval of the check. For more information, see:
+    ## https://docs.datadoghq.com/developers/write_agent_check/#collection-interval
+    #
+    # min_collection_interval: 15
```
Comment (djmitche) [inline pkg/config/config_template.yaml:675]: It looks like this got documented twice, and `discovery_allowed_failures` not documented?

**Review by djmitche — CHANGES_REQUESTED** [review]

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (AlexandreYang) [reply pkg/config/config_template.yaml:675]: @djmitche `discovery_allowed_failures` is already documented in `snmp_listener.discovery_allowed_failures`, but we didn't declared it as known config, an oversight from [previous PR](https://github.com/DataDog/datadog-agent/pull/8227).

There are two different `min_collection_interval`:
- `snmp_listener.min_collection_interval`
- `snmp_listener.configs[].min_collection_interval`

Comment (djmitche) [reply pkg/config/config_template.yaml:675]: Ah, got it.  And apologies for missing the difference in indentation :)

**Review by djmitche — APPROVED** [review]

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (fanny-jiang) [reply pkg/config/config_template.yaml:675]: Perhaps we can add a note to the `snmp_listener.configs[].min_collection_interval` description clarifying that this config is specific to each instance in `snmp_listener.configs[]`.

**Review by fanny-jiang — CHANGES_REQUESTED** [review]

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (AlexandreYang) [reply pkg/config/config_template.yaml:680]: @fanny-jiang Sure, thanks for the review.

Added some precisions here: https://github.com/DataDog/datadog-agent/pull/8880/commits/d354acdf4abab9a17f0c1a943fc1f9e90f693620

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -543,6 +543,21 @@ api_key:
...
+  ## @param loader - string - optional - default: python
+  ## Check loader to use. Available loaders:
+  ## - core: (recommended) Uses new corecheck SNMP integration
+  ## - python: Uses legacy python SNMP integration
+  #
+  # loader: core
+
+  ## @param min_collection_interval - number - optional - default: 15
+  ## This changes the collection interval for the check instances created
+  ## from discovered snmp devices.
```
Comment (alai97) [inline pkg/config/config_template.yaml:555]:
```suggestion
  ## from discovered SNMP devices.
```

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -655,6 +670,15 @@ api_key:
     #
     # loader: core
 
+    ## @param min_collection_interval - number - optional - default: 15
+    ## This changes the collection interval for the check instances created from
+    ## discovered snmp devices. It applies to each specific config from `snmp_listener.configs`
```
Comment (alai97) [inline pkg/config/config_template.yaml:675]:
```suggestion
    ## discovered SNMP devices. It applies to each specific config from `snmp_listener.configs`
```

**Review by alai97 — APPROVED** [review]
Looks good for docs! Just two capitalization suggestions.

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**